app/common/directives/file-upload.directive.js

```diff
@@ -7,29 +7,42 @@ function FileUpload() {
         replace: true,
         scope: {
             container: '=',
-            model: '='
+            model: '=',
+            validation: '='
         },
         controller: [
-            '$scope', '$attrs',
+            '$scope', '$attrs', 'Notify',
             function (
-                $scope, $attrs
+                $scope, $attrs, Notify
             ) {
                 $scope.required = typeof $attrs.required !== 'undefined';
                 $scope.uploadFile = function ($event) {
-                    $scope.container.file = $event.target.files[0];
-                    var reader = new FileReader();
-                    reader.onload = function () {
-                        var dataURL = reader.result;
-                        $scope.container.dataURI = dataURL;
-                        $scope.container.changed = true;
-                        $scope.container.deleted = false;
-                        $scope.model = 'changed';
-                        $scope.$apply();
-                    };
-                    reader.readAsDataURL($event.target.files[0]);
-
-
+                    if (validateFile($event.target.files[0])) {
+                        $scope.container.file = $event.target.files[0];
+                        var reader = new FileReader();
+                        reader.onload = function () {
+                            var dataURL = reader.result;
+                            $scope.container.dataURI = dataURL;
+                            $scope.container.changed = true;
+                            $scope.container.deleted = false;
+                            $scope.model = 'changed';
+                            $scope.$apply();
+                        };
+                        reader.readAsDataURL($event.target.files[0]);
+                    } else {
+                        Notify.error('post.media.error_in_upload');
+                    }
                 };
+
+                function validateFile(container) {
+                    if ($scope.validation === 'image') {
+                        var mimeReg = /[\/.](gif|jpg|jpeg|png)$/i;
+                        var mimeCheck = mimeReg.test(container.type);
+                        var sizeCheck = container.size < 1000000;
```

Comment:
1048576 this is the currently set size in the API. 

Also thinking about it, it's actually really problematic that we don't have a  media_max_size coming from the config itself in features, because if we change the API for an environment this has to be changed as well. 
I think it's fine for this specific fix but for the next version you are working on, I'd suggest adding this as a /config/features item as well, If you agree (I can probably add it quickly in the backend)

Reply:
Yes, i think that is a good idea to fetch it from the backend, lets work on that next?

Reply:
excellent